### PR TITLE
(DOCSP-13018): Remove explicit type mentions in TypeScript Docs

### PR DESCRIPTION
**Pull request by mongomoe**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13018

### Docs staging link (requires sign-in on MongoDB Corp SSO):
* [Node] QuickStart: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/quick-start.html
* [Node] Initialize the Realm App Client: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/init-realmclient.html
* [React Native] Initialize the Realm App Client: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/react-native/init-realmclient.html
* [Node] Manage Email/Password Users: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/manage-email-password-users.html
* [React Native] Manage Email/Password Users: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/react-native/manage-email-password-users.html
* [Node] Create & Manage User API Keys: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/create-manage-api-keys.html
* [React Native] Create & Manage User API Keys: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/react-native/create-manage-api-keys.html
* [Node] Work with Multiple Users: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/work-with-multiple-users.html
* [React Native] Work with Multiple Users: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/react-native/work-with-multiple-users.html
* [Node] MongoDB Data Access: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/mongodb.html
* [Node] Authenticate a User: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/node/authenticate.html
* [React Native] Authenticate a User: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-13018-remove-explicit-types-in-ts-docs/react-native/authenticate.html

### Passing Tests:
MONGODB-TS-TESTS
AUTHENTICATE-TS-TESTS

### CODE REVIEW

File: source/examples/MultiUser/ListUsers/ListUsers.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 // Get a list of all Users
-app.allUsers.forEach((user: Realm.User) => {
+app.allUsers.forEach(user => {
```
Comment (mongomoe) [inline source/examples/MultiUser/ListUsers/ListUsers.ts:2]:
```suggestion
app.allUsers.forEach((user: Realm.User) => {
```

File: source/node/create-manage-api-keys.txt [hunk 2]
```diff
@@ -53,8 +53,8 @@ user's API keys.
 
       .. code-block:: typescript
          
-         const user: Realm.User = app.currentUser;
-         const key: Realm.Auth.ApiKey = await user.apiKeys.create("apiKeyName");
+         const user = app.currentUser;
+         const key = await user.apiKeys.create("apiKeyName");
```
Comment (cbush) [inline source/node/create-manage-api-keys.txt:57]: Why not take this opportunity to migrate these to tests?

Comment (mongomoe) [reply source/node/create-manage-api-keys.txt:57]: Scope creep from the original goal & priority of this ticket. Happy to do that as a separate ticket for the next sprint @cbush

File: examples/node/Examples/mongodb.ts [hunk 3]
```diff
@@ -38,10 +38,10 @@ async function getPlantsCollection() {
...
   // :code-block-end:
   // :code-block-start: plants-collection-handle
-  const mongodb: Realm.Services.MongoDB = app.currentUser.mongoClient(
+  const mongodb = app.currentUser.mongoClient(
     "mongodb-atlas"
   );
-  const plants: MongoDBCollection<Plant> = mongodb
+  const plants = mongodb
     .db("example")
     .collection("plants");
```
Comment (nirinchev) [inline examples/node/Examples/mongodb.ts:46]: This should specify the generic argument to provide the shape of the returned objects.

File: source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
 // Create an email/password credential
 const credentials = Realm.Credentials.emailPassword(
-  "joe.jasper@example.com",
+  // "joe.jasper@example.com",
```
Comment (nirinchev) [inline source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts:3]: Should this be commented out?

**Review by nirinchev — COMMENTED** [review]
Looks good, just a few minor comments

File: examples/node/Examples/mongodb.ts (hunk 3 above)
Comment (nlarew) [reply examples/node/Examples/mongodb.ts:46]:
```suggestion
    .collection<Plant>("plants");
```
^ Per Nikola's suggestion

File: source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts (hunk 4 above)
Comment (mongomoe) [reply source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts:3]: It should not. It's a generated file, looks like there's a known bug in our "Bluehawk" parser. I'll fix this on the generated file temporarily so it still looks good to the docs readers.

Comment (mongomoe) [inline source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts:3]:
```suggestion
   "joe.jasper@example.com",
```

**Review by nirinchev — APPROVED** [review]

File: source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts (hunk 4 above)
Comment (mongomoe) [reply source/examples/generated/code/start/authenticate.codeblock.email-password-login.ts]: Fixed now

**Pull request merged**